Referenced callback fails with NullPointerException

If I declare a callback in my application like this:
```java
@OpenAPIDefinition(
        info = @Info(title = "Greetings", version = "0.0.1"), components = @Components(
                callbacks = @Callback(
                        name = "myEvent",
                        callbackUrlExpression = "{$request.body#/callbackUrl}",
                        operations = @CallbackOperation(
                                method = "POST",
                                responses = {
                                        @APIResponse(responseCode = "2XX", description = "Ok"),
                                        @APIResponse(responseCode = "5XX", description = "Failed"),
                                }
                        )
                )
        )
)
public class GreetingApplication extends Application {
```
it then appears in the generated OpenApi definition as expected:
```yaml
components:
  callbacks:
    myEvent:
      '{$request.body#/callbackUrl}':
        post:
          responses:
            "2XX":
              description: Ok
            "5XX":
              description: Failed
```

However, when I then try to use it, like:
```java
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    @Callback(ref = "#/components/callbacks/myEvent")
    public String hello() {
```

Smallrye OpenApi fails with a NullPointerException:
```
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.Map.keySet()" because the return value of "org.eclipse.microprofile.openapi.models.callbacks.Callback.getPathItems()" is null
	at io.smallrye.openapi.api.util.FilterUtil.filterCallback(FilterUtil.java:218)
	at io.smallrye.openapi.api.util.FilterUtil.filter(FilterUtil.java:132)
	at io.smallrye.openapi.api.util.FilterUtil.filterOperation(FilterUtil.java:261)
	at io.smallrye.openapi.api.util.FilterUtil.filterPathItem(FilterUtil.java:241)
	at io.smallrye.openapi.api.util.FilterUtil.filter(FilterUtil.java:132)
	at io.smallrye.openapi.api.util.FilterUtil.filter(FilterUtil.java:58)
	at io.smallrye.openapi.api.util.FilterUtil.applyFilter(FilterUtil.java:51)
	at io.smallrye.openapi.api.OpenApiDocument.filterModel(OpenApiDocument.java:177)
	at io.smallrye.openapi.api.OpenApiDocument.initialize(OpenApiDocument.java:134)
	at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.loadDocument(SmallRyeOpenApiProcessor.java:1135)
	at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.build(SmallRyeOpenApiProcessor.java:855)
```

Now, `CallbackItem#pathItems` is initially `null`, and during initialization of the referencing callback, `CallbackImpl#addPathItem` is called with `null` for both the `item` and `name` parameter. Doing so still leaves `CallbackItem#pathItems` as `null`.

I've come to believe that `CallbackItem#getPathItems` is allowed to return `null`, as the default `hasPathItem` method in `org.eclipse.microprofile.openapi.models.callbacks.Callback` looks like this:
```java
    default boolean hasPathItem(String name) {
        Map<String, PathItem> map = this.getPathItems();
        return map == null ? false : map.containsKey(name);
    }
```

This must mean that this code in `FilterUtil` incorrectly assumes that `CallbackItem#getPathItems` won't return `null`:
https://github.com/smallrye/smallrye-open-api/blob/42c19ecf7f6bfa347288a51ec24a85733ae923f7/core/src/main/java/io/smallrye/openapi/api/util/FilterUtil.java#L218

It works perfectly if i change `CallbackItem#pathItems` to an empty Map with a debugger.
Tested with Quarkus 3.6.0 using `io.smallrye:smallrye-open-api-core:3.7.0`.

--- a/core/src/main/java/io/smallrye/openapi/api/util/FilterUtil.java
+++ b/core/src/main/java/io/smallrye/openapi/api/util/FilterUtil.java
@@ -6,9 +6,11 @@ import java.util.IdentityHashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.function.BiConsumer;
 import java.util.function.Consumer;
 import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
 
 import org.eclipse.microprofile.openapi.OASFilter;
 import org.eclipse.microprofile.openapi.models.Components;
@@ -214,17 +216,20 @@ public class FilterUtil {
      * @param model
      */
     private void filterCallback(OASFilter filter, Callback model) {
-        if (model != null) {
-            Collection<String> keys = new ArrayList<>(model.getPathItems().keySet());
-            for (String key : keys) {
-                PathItem childModel = model.getPathItem(key);
-                filterPathItem(filter, childModel);
+        Optional.ofNullable(model)
+                .map(Callback::getPathItems)
+                .map(Map::keySet)
+                .map(ArrayList::new)
+                .map(Collection::stream)
+                .orElseGet(Stream::empty)
+                .forEach(key -> {
+                    PathItem childModel = model.getPathItem(key);
+                    filterPathItem(filter, childModel);
 
-                if (filter.filterPathItem(childModel) == null) {
-                    model.removePathItem(key);
-                }
-            }
-        }
+                    if (filter.filterPathItem(childModel) == null) {
+                        model.removePathItem(key);
+                    }
+                });
     }
 
     /**

--- a/extension-jaxrs/src/test/java/io/smallrye/openapi/runtime/scanner/CallbackScanTests.java
+++ b/extension-jaxrs/src/test/java/io/smallrye/openapi/runtime/scanner/CallbackScanTests.java
@@ -1,0 +1,50 @@
+package io.smallrye.openapi.runtime.scanner;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.Application;
+import jakarta.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.openapi.OASConfig;
+import org.eclipse.microprofile.openapi.annotations.Components;
+import org.eclipse.microprofile.openapi.annotations.OpenAPIDefinition;
+import org.eclipse.microprofile.openapi.annotations.callbacks.Callback;
+import org.eclipse.microprofile.openapi.annotations.callbacks.CallbackOperation;
+import org.eclipse.microprofile.openapi.annotations.info.Info;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.jboss.jandex.Index;
+import org.junit.jupiter.api.Test;
+
+import io.smallrye.openapi.api.util.UnusedSchemaFilter;
+import io.smallrye.openapi.runtime.OpenApiProcessor;
+
+class CallbackScanTests extends IndexScannerTestBase {
+
+    @Test
+    void testCallbackReference() throws Exception {
+        @OpenAPIDefinition(info = @Info(title = "Greetings", version = "0.0.1"), components = @Components(callbacks = @Callback(name = "myEvent", callbackUrlExpression = "{$request.body#/callbackUrl}", operations = @CallbackOperation(method = "POST", responses = {
+                @APIResponse(responseCode = "2XX", description = "Ok"),
+                @APIResponse(responseCode = "5XX", description = "Failed"),
+        }))))
+        class App extends Application {
+
+        }
+
+        @Path("/")
+        class Resource {
+            @GET
+            @Produces(MediaType.TEXT_PLAIN)
+            @Callback(ref = "#/components/callbacks/myEvent")
+            public String hello() {
+                return "goodbye";
+            }
+        }
+
+        Index index = indexOf(App.class, Resource.class);
+        OpenAPI result = OpenApiProcessor.bootstrap(dynamicConfig(OASConfig.FILTER, UnusedSchemaFilter.class.getName()), index);
+        printToConsole(result);
+        assertJsonEquals("callback.reference-component-callback.json", result);
+    }
+}

--- a/extension-jaxrs/src/test/resources/io/smallrye/openapi/runtime/scanner/callback.reference-component-callback.json
+++ b/extension-jaxrs/src/test/resources/io/smallrye/openapi/runtime/scanner/callback.reference-component-callback.json
@@ -1,0 +1,48 @@
+{
+  "openapi" : "3.0.3",
+  "info" : {
+    "title" : "Greetings",
+    "version" : "0.0.1"
+  },
+  "paths" : {
+    "/" : {
+      "get" : {
+        "responses" : {
+          "200" : {
+            "description" : "OK",
+            "content" : {
+              "text/plain" : {
+                "schema" : {
+                  "type" : "string"
+                }
+              }
+            }
+          }
+        },
+        "callbacks" : {
+          "myEvent" : {
+            "$ref" : "#/components/callbacks/myEvent"
+          }
+        }
+      }
+    }
+  },
+  "components" : {
+    "callbacks" : {
+      "myEvent" : {
+        "{$request.body#/callbackUrl}" : {
+          "post" : {
+            "responses" : {
+              "2XX" : {
+                "description" : "Ok"
+              },
+              "5XX" : {
+                "description" : "Failed"
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}

